### PR TITLE
Allow multiple workers by moving from pub/sub to lpush/blpop.

### PR DESCRIPTION
**Pull request by sfrycertinia**

### CODE REVIEW

File: src/lib/index/shared/configValidator.js [hunk 1]
```diff
@@ -35,6 +35,9 @@ const
 		if (config.url == null || !_.isString(config.url)) {
 			throw new Error('Invalid parameter: url not a string.');
 		}
+		if (!config.return_buffers) {
```
Comment (dfruddffdc) [inline src/lib/index/shared/configValidator.js]: Why have the property at all, if it always errors when the property is falsey?

File: src/lib/index/shared/redis.js [hunk 2]
```diff
@@ -81,38 +78,56 @@ const
...
-		const redisInstance = getOrCreateRedisInstance(config);
+		const processMessage = () => {
 
-		connection.subscribe(channel);
-		redisInstance.subscribersByChannel.set(channel, handler);
+			connection.blpop(channel, 0, (err, data) => {
+				if (err) {
+					throw err;
+				}
+				handler(data[1]);
```
Comment (dfruddffdc) [inline src/lib/index/shared/redis.js]: What's the significance of data[1]? Maybe worth an explanatory comment.

**Review by dfruddffdc — DISMISSED** [review]

File: src/lib/index/shared/redis.js (hunk 2 above)
Comment (sfrycertinia) [reply src/lib/index/shared/redis.js]: Good point. It is a rather interesting signature, the channel is the first entry in the array and the data the second.

File: src/lib/index/shared/configValidator.js (hunk 1 above)
Comment (sfrycertinia) [reply src/lib/index/shared/configValidator.js]: It's a Redis setting. I'll move the setting of the config option to the point that we call into Redis and remove from our validation.

**Review by dfruddffdc — APPROVED** [review]

**Pull request merged**